src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -371,26 +371,79 @@ public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldTyp
         _dataIndex = dataLocation;
     }
 
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private int WriteNullsTableIfRequired<TEnumerator>(TEnumerator values, int dataLocation, ref IndexEntryFieldType indexEntryFieldLocation)
         where TEnumerator : IReadOnlySpanIndexer
+    {        
+        // If we have an small number of values, we just allocate the space for the nulls table in the stack. 
+        if (values.Length < 256)
```

Comment:
This should be `< (256*8)`, instead.
You aren't checking values, but the size is in bits.

Reply:
Actually that was `32 * 8 = 256`. Allocating 32 bytes here.